### PR TITLE
Adds generate_form function to the Shipment object

### PR DESCRIPTION
**Pull request by jchen293**
# Description

Adds `generate_form` function to the Shipment object

# Testing

Added a new test and passed

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

File: easypost/shipment.py [hunk 1]
```diff
@@ -74,6 +74,18 @@ def lowest_smartrate(self, delivery_days: int, delivery_accuracy: str) -> Rate:
 
         return lowest_smartrate
 
+    def generate_form(self, form_type, form_options={}) -> "Shipment":
```
Comment (nwithan8) [inline easypost/shipment.py:77]: Please add type hints for the parameters
```suggestion
    def generate_form(self, form_type: str, form_options: dict = {}) -> "Shipment":
```

**Review by nwithan8 — DISMISSED** [review]

File: easypost/shipment.py (hunk 1 above)
Comment (Justintime50) [reply easypost/shipment.py:77]: Yeah @jchen293 can we please add type hints here per Nate's suggestion? The `dict` will need to be `Dict` and it should be `Optional[Dict[str, any]]`

Comment (jchen293) [reply easypost/shipment.py]: resolved

**Review by Justintime50 — DISMISSED** [review]

**Comment by nwithan8** [comment]
Lint still failing.

**Comment by nwithan8** [comment]
@jchen293 slap a `# type: ignore` comment on line 83 in `shipments.py`

**Review by nwithan8 — APPROVED** [review]

**Review by Justintime50 — DISMISSED** [review]

**Review by Justintime50 — APPROVED** [review]

**Review by nwithan8 — APPROVED** [review]

**Pull request merged**